File: go-leetcode/binary-tree/145.二叉树的后序遍历.go
package leetcode

func postorderTraversal(root *TreeNode) []int {
	res := []int{}
	var postorder func(node *TreeNode)

	postorder = func(node *TreeNode) {
		if node == nil {
			return
		}
		postorder(node.Left)
		postorder(node.Right)
		res = append(res, node.Val)
	}

	postorder(root)

	return res
}

File: python-leetcode/binary-tree/145.二叉树的后序遍历.py
from typing import List, Optional

from type import TreeNode


class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []

        def postorder(node: Optional[TreeNode]) -> None:
            if not node:
                return
            postorder(node.left)
            postorder(node.right)
            res.append(node.val)

        postorder(root)
        return res
